### PR TITLE
Setup a New Project

### PR DESCRIPTION
**Pull request by ksdenisova**
Acceptance:

- Create GitHub repo (invite team, create Readme, and license)
- Create gitignore for Elixir, Ruby and IDE
- Have Unit and Acceptance tests running

[Trello](https://trello.com/c/zLKL98dQ/2-set-up-a-new-project)

### CODE REVIEW

**Review by yaoandrew — APPROVED** [review]
:shipit: Woohoo! Server time.

File: .gitignore [hunk 1]
```diff
@@ -8,3 +9,57 @@ erl_crash.dump
...
+### VisualStudioCode Patch ###
+# Ignore all local history of files
+.history
+.ionide
+
+# Support for Project snippet scope
+!.vscode/*.code-snippets
+
+## System ##
+*.DS_Store
```
Comment (Aoxian) [inline .gitignore:65]: Have you heard about [.gitignore global](https://gist.github.com/subfuzion/db7f57fff2fb6998a16c) setup?

Most developers choose to use different tools for their local development.  A project's `.gitignore` can get unwieldily very fast if it include each and every IDE's ignores for every developer.  I suggest setting up a `.gitignore` globally once for your tools and it applies to every repo now and in the future you work on in addition to the project `.gitignore`.  This also has the added bonus of the project's `.gitignore` only focusing on important thing for the project (like build directories, or environment variable files).

Comment (yaoandrew) [reply .gitignore:65]: TIL need to update my projects 😅

**Review by Aoxian — COMMENTED** [review]
I suggest not using submodules for the acceptance test repo and instead create the folder (http_server_spec) and only copy in the tests you need as you work on new tickets.

File: .gitignore (hunk 1 above)
Comment (Aoxian) [reply .gitignore:65]: Here is an example of my `.gitignore` global file: https://github.com/Aoxian/dotfiles/blob/master/.gitignore_global

**Comment by ksdenisova** [comment]
> I suggest not using submodules for the acceptance test repo and instead create the folder (http_server_spec) and only copy in the tests you need as you work on new tickets.

I'll change it to the folder, because I'm not sure what happens if I change a file and push.

I have this idea for running test separately:
I can add this section in `Rakefile`:
`desc "Run tests for all implemented features in 01_getting_started"
  task :f0 do
    if is_server_running?(5000)
      puts `bundle exec spinach --tags implemented`
    else
      server_not_running_error()
    end
  end`

Then once I implemented a new feature, I tag it as `implemented`, and run my tests using `rake task:f0`. When all required features was implemented, I run all tests from finished section `rake task:f1` for example, and then to to the next test. 

It will look like this:
CleanShot 2021-11-18 at 14 53 37@2x

File: .gitignore (hunk 1 above)
Comment (ksdenisova) [reply .gitignore:65]: No, I haven't. I'll create one for me now. In this case, in local `.gitignore` file should be listed only ignored files specific for this project? Like Ruby and Elixir related.

Comment (ksdenisova) [reply .gitignore]: Configured a global `.gitignore` and updated local.

**Comment by ksdenisova** [comment]
> I suggest not using submodules for the acceptance test repo and instead create the folder (http_server_spec) and only copy in the tests you need as you work on new tickets.

Created a folder as you suggested and added opportunity to run specific tests.

**Comment by Aoxian** [comment]
> Then once I implemented a new feature, I tag it as `implemented`, and run my tests using `rake test:f0`. When all required features was implemented, I run all tests from finished section `rake test:f1` for example, and then to to the next test.

I like the plan!  Only suggestion is to use a better name here than "f0" and "f1".

**Review by Aoxian — APPROVED** [review]

**Comment by ksdenisova** [comment]
> > Then once I implemented a new feature, I tag it as `implemented`, and run my tests using `rake test:f0`. When all required features was implemented, I run all tests from finished section `rake test:f1` for example, and then to to the next test.
> 
> I like the plan! Only suggestion is to use a better name here than "f0" and "f1".

I'm following the naming that is used in `Rakefile`. Each group of test can be run separately and they named existing tasks: _f1, f2, f3, f4_.

**Comment by Aoxian** [comment]
> > > Then once I implemented a new feature, I tag it as `implemented`, and run my tests using `rake test:f0`. When all required features was implemented, I run all tests from finished section `rake test:f1` for example, and then to to the next test.
> > 
> > 
> > I like the plan! Only suggestion is to use a better name here than "f0" and "f1".
> 
> I'm following the naming that is used in `Rakefile`. Each group of test can be run separately and they named existing tasks: _f1, f2, f3, f4_.

This is no longer a submodule, so you have complete control. Is there another reason perhaps why these names were chosen originally? should we not refactor code because of an established pattern? it is good to think through these types of questions.  IF we can rename these tags to add clarity, perhaps a PR could be open against the repo to update the names. 

I approved the PR so I do not believe this is a blocking issues, just things to think about.

**Comment by ksdenisova** [comment]
> > > > Then once I implemented a new feature, I tag it as `implemented`, and run my tests using `rake test:f0`. When all required features was implemented, I run all tests from finished section `rake test:f1` for example, and then to to the next test.
> > > 
> > > 
> > > I like the plan! Only suggestion is to use a better name here than "f0" and "f1".
> > 
> > 
> > I'm following the naming that is used in `Rakefile`. Each group of test can be run separately and they named existing tasks: _f1, f2, f3, f4_.
> 
> This is no longer a submodule, so you have complete control. Is there another reason perhaps why these names were chosen originally? should we not refactor code because of an established pattern? it is good to think through these types of questions. IF we can rename these tags to add clarity, perhaps a PR could be open against the repo to update the names.
> 
> I approved the PR so I do not believe this is a blocking issues, just things to think about.

I think they were chosen only for simplicity. We can see that in original repo authors add comments about what is going to run for every command:
````
rake test:f1 # Run all of the tests in 01_getting_started
rake test:f2 # Run all of the tests in 02_structured_data
rake test:f3 # Run all of the tests in 03_file_server
rake test:f4 # Run all of the tests in 04_todo_list
````

But I agree we can use more explicit names. And when if was a submodule, git couldn't see the changes in files. I have never worked with submodules, so it was a good example for me. I'll add this update to the Backlog as a separate PR.

**Pull request merged**